### PR TITLE
Add Elasticsearch Low Level Rest Client

### LINKED ISSUE #5736: Add Elasticsearch Low Level Rest Client
### Describe the feature here

Add an extension for the component

### PR DESCRIPTION
**Pull request by zbendhiba**
Fixes #5736

I created a follow up issue to add more test coverage.  https://github.com/apache/camel-quarkus/issues/5737

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by zbendhiba** [comment]
The test failure seems unrelated. Is master-openshift considered a flaky test ?

File: integration-tests-jvm/elasticsearch-rest-client/src/test/java/org/apache/camel/quarkus/component/elasticsearch/rest/client/it/ElasticsearchTestResource.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                    .withEnv("discovery.type", "single-node")
+                    .withEnv("xpack.security.enabled", "true")
+                    .withEnv("action.destructive_requires_name", "false") // needed for deleting all indexes after each test (allowing _all wildcard)
+                    .withEnv("ELASTIC_USERNAME", ELASTICSEARCH_USERNAME)
+                    .withEnv("ELASTIC_PASSWORD", ELASTICSEARCH_PASSWORD)
+                    .waitingFor(Wait.forListeningPort());
+
+            container.start();
+
+            String hostAddresses = String.format("localhost:%s", container.getMappedPort(ELASTICSEARCH_PORT));
```
Comment (jamesnetherton) [inline integration-tests-jvm/elasticsearch-rest-client/src/test/java/org/apache/camel/quarkus/component/elasticsearch/rest/client/it/ElasticsearchTestResource.java:59]:
```suggestion
            String hostAddresses = String.format("%s:%s",  container.getHost(), container.getMappedPort(ELASTICSEARCH_PORT));
```

**Comment by jamesnetherton** [comment]
> The test failure seems unrelated. Is master-openshift considered a flaky test ?

It got enabled yesterday after being disabled for a long time. I will try to take a look into it.

**Comment by zbendhiba** [comment]
thanks @jamesnetherton !

File: integration-tests-jvm/elasticsearch-rest-client/src/main/java/org/apache/camel/quarkus/component/elasticsearch/rest/client/it/ElasticsearchRestClientResource.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    @Inject
+    CamelContext context;
+
+    @Inject
+    ProducerTemplate producerTemplate;
+
+    @Path("/load/component/elasticsearch-rest-client")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response loadComponentElasticsearchRestClient() throws Exception {
```
Comment (aldettinger) [inline integration-tests-jvm/elasticsearch-rest-client/src/main/java/org/apache/camel/quarkus/component/elasticsearch/rest/client/it/ElasticsearchRestClientResource.java:50]: Should we remove the auto-generated resource / test ? Like if index test is passing, we know for the sure the component can be loaded  ?

**Review by aldettinger — APPROVED** [review]
LGTM, the failure is probably related to re-formatting need only.

@zbendhiba If one is currently implementing an example that transform data and load it into ElasticSearch. Would you advise to use this low level rest client ? or camel-quarkus-elasticsearch ?

File: integration-tests-jvm/elasticsearch-rest-client/src/main/java/org/apache/camel/quarkus/component/elasticsearch/rest/client/it/ElasticsearchRestClientResource.java (hunk 2 above)
Comment (zbendhiba) [reply integration-tests-jvm/elasticsearch-rest-client/src/main/java/org/apache/camel/quarkus/component/elasticsearch/rest/client/it/ElasticsearchRestClientResource.java:50]: yes I'll add that to the follow up issue

**Comment by zbendhiba** [comment]
I pushed again because of format error and squashed commits

**Pull request merged**